packages/@aws-cdk/aws-codebuild/lib/codepipeline-source.ts

```diff
@@ -0,0 +1,15 @@
+import { Source } from './source';
+import { CODEPIPELINE_SOURCE_ARTIFACTS_TYPE } from './source-types';
+
+/**
+ * CodePipeline Source definition for a CodeBuild Project.
+ * *Note*: this type cannot be used as a secondary source,
+ * and because of that, you're not allowed to specify an identifier for it.
+ */
+export class CodePipelineSource extends Source {
```

Comment:
Ditto, should be accessible through `codebuild.Source.codePipeline()` and internal

Reply:
Same as above - this Source type is actually private to the CodeBuild package, and not available to customers at all.

Reply:
Not sure I understand... Do we only allow people to use the high level `PipelineProject` in order to set up a codebuild into a pipeline. Doesn't sound right. I don't see a problem exposing `Source.codePipeline` (and TBH, `Source.none` as well...